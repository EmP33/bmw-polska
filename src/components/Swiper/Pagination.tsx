import React from "react";
// Styles
import { Pagination } from "./Swiper.styles";

interface Props {
  swipeSlide: number;
  handleChangeSlide: (numb: number) => void;
}

const PaginationComponent: React.FC<Props> = ({
  swipeSlide,
  handleChangeSlide,
}) => {
  return (
    <Pagination>
      <label className="container">
        <input
          data-testid="pag1"
          type="radio"
          name="pagination"
          id="pag1"
          onChange={handleChangeSlide.bind(null, 0)}
          checked={swipeSlide === 0}
        />
        <span className="checkmark"></span>
      </label>
      <label className="container">
        <input
          data-testid="pag2"
          type="radio"
          name="pagination"
          id="pag2"
          onChange={handleChangeSlide.bind(null, 1)}
          checked={swipeSlide === 1}
        />
        <span className="checkmark"></span>
      </label>
      <label className="container">
        <input
          data-testid="pag3"
          type="radio"
          name="pagination"
          id="pag3"
          onChange={handleChangeSlide.bind(null, 2)}
          checked={swipeSlide === 2}
        />
        <span className="checkmark"></span>
      </label>
    </Pagination>
  );
};

export default PaginationComponent;
