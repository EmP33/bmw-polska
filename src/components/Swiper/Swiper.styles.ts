import styled from "styled-components";

export const Swiper = styled.div<{ swipeSlide: number }>`
  height: 100%;
  width: 300vw;
  color: #fff;
  display: flex;
  transform: ${(props) =>
    props.swipeSlide === 0
      ? "translateX(0)"
      : props.swipeSlide === 1
      ? "translateX(-100vw)"
      : "translateX(-200vw)"};
  transition: all 0.3s linear;

  .slide {
    width: 100vw;
    img {
      height: 100%;
      object-position: center;

      @media only screen and (max-width: 400px) {
        width: auto;
      }
      @media only screen and (min-width: 2150px) {
        width: 100%;
      }
    }
  }
`;

export const Pagination = styled.ul`
  position: absolute;
  left: 0;
  bottom: 12%;
  list-style: none;
  display: flex;
  column-gap: 5rem;
  width: 225px;

  @media only screen and (max-width: 1200px) {
    bottom: 5%;
    left: 50%;
    transform: translateX(-50%);
  }

  /* The container */
  .container {
    display: block;
    position: relative;
    cursor: pointer;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
  }

  /* Hide the browser's default checkbox */
  .container input {
    display: none;
  }

  /* Create a custom checkbox */
  .checkmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 5px;
    width: 3.75rem;
    background-color: var(--color-white);
  }

  /* On mouse-over, add a grey background color */
  .container:hover input ~ .checkmark {
    background-color: var(--color-gray-6);
  }

  /* When the checkbox is checked, add a blue background */
  .container input:checked ~ .checkmark {
    background-color: var(--color-blue);
  }
`;
