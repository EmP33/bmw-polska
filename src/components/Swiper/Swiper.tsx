import React from "react";

// Styles
import { Swiper } from "./Swiper.styles";

// Images
import SlideImg from "../../assets/images/slider1.webp";
import MobileSlideImg from "../../assets/images/slider1-mobile.webp";

// Hooks
import useMediaQuery from "../../hooks/useMediaQuery";

interface Props {
  swipeSlide: number;
}

const SwiperComponent: React.FC<Props> = ({ swipeSlide }) => {
  const matches = useMediaQuery("(min-width: 401px)");
  return (
    <Swiper swipeSlide={swipeSlide}>
      <div className="slide">
        <img src={matches ? SlideImg : MobileSlideImg} alt="New BMW iX slide" />
      </div>
      <div className="slide">
        <img src={matches ? SlideImg : MobileSlideImg} alt="New BMW iX slide" />
      </div>
      <div className="slide">
        <img src={matches ? SlideImg : MobileSlideImg} alt="New BMW iX slide" />
      </div>
    </Swiper>
  );
};

export default SwiperComponent;
