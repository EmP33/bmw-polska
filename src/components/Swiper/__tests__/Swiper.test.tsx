import Swiper from "../Swiper";
import ReactShallowRenderer from "react-test-renderer/shallow";
import renderer from "react-test-renderer";
import "jest-styled-components";
import styled from "styled-components";
import { render, screen } from "@testing-library/react";

const StyledSwiper = styled.div<{ swipeSlide: number }>`
  height: 100%;
  width: 300vw;
  color: #fff;
  display: flex;
  transform: ${(props) =>
    props.swipeSlide === 0
      ? "translateX(0)"
      : props.swipeSlide === 1
      ? "translateX(-100vw)"
      : "translateX(-200vw)"};
  transition: all 0.3s linear;

  .slide {
    width: 100vw;
    img {
      height: 100%;
      object-position: center;
    }
  }
`;

describe("<Swiper />", () => {
  const renderProps = (matches?: boolean = false) => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: matches, // matches
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  };
  it("should render Swiper Component properly", () => {
    renderProps();
    const renderer = new ReactShallowRenderer();
    renderer.render(<Swiper swipeSlide={0} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("should have transform translateX(0) style when swipeSlide is 0", () => {
    renderProps();
    const swiper = renderer.create(<StyledSwiper swipeSlide={0} />).toJSON();
    expect(swiper).toHaveStyleRule("transform", "translateX(0)");
  });

  it("should have transform translateX(-100vw) style when swipeSlide is 1", () => {
    renderProps();
    const swiper = renderer.create(<StyledSwiper swipeSlide={1} />).toJSON();
    expect(swiper).toHaveStyleRule("transform", "translateX(-100vw)");
  });

  it("should have transform translateX(-200vw) style when swipeSlide is 2", () => {
    renderProps();
    const swiper = renderer.create(<StyledSwiper swipeSlide={2} />).toJSON();
    expect(swiper).toHaveStyleRule("transform", "translateX(-200vw)");
  });

  it("should use mobile image when screen resolution is less than 360px", () => {
    renderProps();
    render(<Swiper swipeSlide={0} />);
    const imageElements = screen.getAllByAltText(
      "New BMW iX slide"
    ) as HTMLImageElement[];

    expect(imageElements[0].src).toContain("slider1-mobile.png");
    expect(imageElements[1].src).toContain("slider1-mobile.png");
    expect(imageElements[2].src).toContain("slider1-mobile.png");
  });

  it("should use normal image when screen resolution is greater than 360px", () => {
    renderProps(true);
    render(<Swiper swipeSlide={0} />);
    const imageElements = screen.getAllByAltText(
      "New BMW iX slide"
    ) as HTMLImageElement[];

    expect(imageElements[0].src).toContain("slider1.png");
    expect(imageElements[1].src).toContain("slider1.png");
    expect(imageElements[2].src).toContain("slider1.png");
  });
});
