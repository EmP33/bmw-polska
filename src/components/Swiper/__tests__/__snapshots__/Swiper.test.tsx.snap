// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Swiper /> should render Swiper Component properly 1`] = `
<styled.div
  swipeSlide={0}
>
  <div
    className="slide"
  >
    <img
      alt="New BMW iX slide"
      src="slider1-mobile.webp"
    />
  </div>
  <div
    className="slide"
  >
    <img
      alt="New BMW iX slide"
      src="slider1-mobile.webp"
    />
  </div>
  <div
    className="slide"
  >
    <img
      alt="New BMW iX slide"
      src="slider1-mobile.webp"
    />
  </div>
</styled.div>
`;
